sdk/core/core-http-compat/test/cloneRequestPolicy.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  CompatResponse,
+  RequestPolicy,
+  RequestPolicyFactory,
+  RequestPolicyOptionsLike,
+  WebResourceLike,
+} from "../src/index";
+
+export function cloneRequestPolicy(): RequestPolicyFactory {
+  return {
+    create: (nextPolicy: RequestPolicy, options: RequestPolicyOptionsLike) => {
+      return new CloneRequestPolicy(nextPolicy, options);
+    },
+  };
+}
+
+export class CloneRequestPolicy {
+  constructor(private _nextPolicy: RequestPolicy, _options: RequestPolicyOptionsLike) {
```

Comment:
Kind of weirds me out that we accept some unvisited options in the construction of this shim.

Reply:
you mean that _options never gets used? 

This is a particularly obnoxious facet of core-http request policies (https://github.com/Azure/azure-sdk-for-js/blob/92116358d99ec4eb3d8a0e62ed49a09dc4a0ee18/sdk/core/core-http/src/policies/requestPolicy.ts#L13) whereby there is a forced "common" options bag that essentially no policy used all because of ancient logging stuff that I think was left in from the ms-rest-js days: https://github.com/Azure/azure-sdk-for-js/blob/92116358d99ec4eb3d8a0e62ed49a09dc4a0ee18/sdk/core/core-http/src/policies/requestPolicy.ts#L56